docs/domain/03_aggregates.md

```diff
@@ -0,0 +1,67 @@
+# Aggregates
+
+Aggregates are part of DDD. To create one you have to create entity class inheriting [`IAggregateRoot`](../../src/Domain/LeanCode.DomainModels/Model/IAggregateRoot.cs) or [`IAggregateRootWithoutOptimisticConcurrency`](../../src/Domain/LeanCode.DomainModels/Model/IAggregateRoot.cs).
+
+## [`IAggregateRoot`](../../src/Domain/LeanCode.DomainModels/Model/IAggregateRoot.cs)
+
+In order to create aggregate root you should create class inheriting it. [`IAggregateRoot`](../../src/Domain/LeanCode.DomainModels/Model/IAggregateRoot.cs) requires you to specify Id type.
+
+Consider the following aggregate.
+
+```csharp
+public class User : IAggregateRoot<UserId>
+{
+    public UserId Id { get; private init; }
+    public string Name { get; private init; } = null!;
+
+    DateTime IOptimisticConcurrency.DateModified { get; set; }
+
+    private User()
+    { }
+
+    public User(UserId id, string name)
+    {
+        Id = id;
+        Name = name;
+    }
+}
+```
+
+### Parameterless constructor
+
+Parameterless constructor is required by entity framework. It is private to not be accessible outside.
```

Comment:
```suggestion
Parameterless constructor is required by EntityFramework. It is private because it should not be used in other scenarios, especially application code.
```